include/mqtt/server.hpp

```diff
@@ -38,6 +38,7 @@ class server_endpoint : public endpoint<Mutex, LockGuard, PacketIdBytes> {
 public:
     using endpoint<Mutex, LockGuard, PacketIdBytes>::endpoint;
 protected:
+    bool on_v5_connack(bool, v5::connect_reason_code, v5::properties) noexcept override { return true; }
```

Comment:
Necessary to add? I don't see a new pure-virtual function added to endpoint.hpp ?

Reply:
Yes. It is for the existing pure-vritual function at endpoint.hpp. (MQTT v5 CONNACK handler).

However, it depends on [the client side session expiry process](https://github.com/redboltz/mqtt_cpp/pull/657/files#r476813114).

If the client side session expiry process is required, the client need to know the value of SessionExpiryInterval.

In the following case, the value is notified by the broker.

https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901082

> If the Session Expiry Interval is absent the value in the CONNECT Packet used. The server uses this property to inform the Client that it is using a value other than that sent by the Client in the CONNACK. Refer to section 3.1.2.11.2 for a description of the use of Session Expiry Interval.

So the client needs to store the value.
See https://github.com/redboltz/mqtt_cpp/pull/657/files#diff-3d4267aa87c422df49b85a5e6fe94072R1450

callable_overlay call the base class of on_v5_connack.
https://github.com/redboltz/mqtt_cpp/pull/657/files#diff-456c816ee881b91e76edab876a15542eR359

It causes link error at the server. (Note: Both client and server use callable_overlay)

It is the same pattern as https://github.com/redboltz/mqtt_cpp/pull/657/files#diff-dfd5df761d5065103d919d255c1389b9R42

Reply:
Ok. Makes sense.